File: src/main/java/net/thedragonteam/armorplus/items/base/BaseDevItem.java
package net.thedragonteam.armorplus.items.base;

import net.minecraft.creativetab.CreativeTabs;
import net.minecraft.item.ItemStack;
import net.minecraft.util.NonNullList;
import net.minecraftforge.fml.relauncher.Side;
import net.minecraftforge.fml.relauncher.SideOnly;
import net.thedragonteam.armorplus.ArmorPlus;
import net.thedragonteam.armorplus.iface.IModelHelper;
import net.thedragonteam.armorplus.items.enums.DevItems;

import static net.thedragonteam.thedragonlib.util.ItemStackUtils.getItemStack;

/**
 * @author Sokratis Fotkatzikis - TheDragonTeam
 **/
public class BaseDevItem extends BaseItem implements IModelHelper {

    private final DevItems devItems;

    public BaseDevItem(DevItems devItems) {
        super(devItems.getName());
        this.devItems = devItems;
        if (devItems.hasSubTypes()) {
            this.setHasSubtypes(true);
            this.setMaxDamage(0);
        }
        this.setCreativeTab(ArmorPlus.tabArmorplusItems);
    }

    @Override
    @SideOnly(Side.CLIENT)
    public void initModel() {
        if (devItems.hasSubTypes()) this.initModel(getRegistryName(), "_second", "cosmetics", 1);
        this.initModel(getRegistryName(), "cosmetics", 0);
    }

    @Override
    public void getSubItems(CreativeTabs tab, NonNullList<ItemStack> items) {
        if (isInCreativeTab(tab)) {
            if (devItems.hasSubTypes()) {
                items.add(getItemStack(this, 0));
                items.add(getItemStack(this, 1));
            }
        }
    }

}

File: src/main/java/net/thedragonteam/armorplus/items/base/BaseDevItem.kt
/*
 * Copyright (c) TheDragonTeam 2016-2017.
 */

package net.thedragonteam.armorplus.items.base

import net.minecraft.creativetab.CreativeTabs
import net.minecraft.item.ItemStack
import net.minecraft.util.NonNullList
import net.minecraftforge.fml.relauncher.Side
import net.minecraftforge.fml.relauncher.SideOnly
import net.thedragonteam.armorplus.ArmorPlus
import net.thedragonteam.armorplus.iface.IModelHelper
import net.thedragonteam.armorplus.items.enums.DevItems
import net.thedragonteam.thedragonlib.util.ItemStackUtils.getItemStack

class BaseDevItem(private val devItems: DevItems) : BaseItem(devItems.getName()), IModelHelper {

    init {
        if (devItems.hasSubTypes()) {
            this.setHasSubtypes(true)
            maxDamage = 0
        }
        this.creativeTab = ArmorPlus.tabArmorplusItems
    }

    @SideOnly(Side.CLIENT)
    override fun initModel() {
        if (devItems.hasSubTypes()) this.initModel(registryName, "_second", "cosmetics", 1)
        this.initModel(registryName, "cosmetics", 0)
    }

    override fun getSubItems(tab: CreativeTabs?, subItems: NonNullList<ItemStack>) {
        if (isInCreativeTab(tab)) {
            if (devItems.hasSubTypes()) {
                subItems.add(getItemStack(this, 0))
                subItems.add(getItemStack(this, 1))
            }
        }
    }

}
